commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/ftp/FtpFileSystemConfigBuilder.java

```diff
@@ -238,6 +242,19 @@ public Boolean getUserDirIsRoot(final FileSystemOptions opts) {
         return getBoolean(opts, USER_DIR_IS_ROOT, Boolean.TRUE);
     }
 
+    /**
+     * @param opts The FileSystem options.
+     * @return The list of status codes (apart from 200) that are considered as OK when prematurely
+     * closing a stream. Defaults to <code>[426, 550]</code>.
+     * @since 2.4
+     */
+    public List<Integer> getTransferAbortedOkStatusCodes(final FileSystemOptions opts) {
+        // See VFS-674, its accompanying PR and https://github.com/apache/commons-vfs/pull/51 as to why 426 and 550 are here
+        @SuppressWarnings("unchecked")
+        List<Integer> statusCodes = (List<Integer>) getParam(opts, TRANSFER_ABORTED_OK_STATUS_CODES);
+        return statusCodes != null ? statusCodes : Arrays.asList(FTPReply.TRANSFER_ABORTED, FTPReply.FILE_UNAVAILABLE);
```

Comment:
@boris-petrov ,

Thank you so much for your PR Boris. I think you need a unit test that sets expectations for how this configuration option works. For example *I* was expecting this to work but it will not:

```getTransferAbortedOkStatusCodes(opts).add(555)```

I think the above should be fixed which implies a different implementation in the builder. Perhaps the builder is initialized with ```Arrays.asList(FTPReply.TRANSFER_ABORTED, FTPReply.FILE_UNAVAILABLE);``` if it is null. IOW, the new array should be saved.

Reply:
@boris-petrov,

One more detail.

I would also create a protected method called ```getDefault...``` that returns the ```Arrays.asList(FTPReply.TRANSFER_ABORTED, FTPReply.FILE_UNAVAILABLE)``` to make it clear what the defaults are with the possibility of easy override though subclassing. 

Gary

Reply:
Thanks for the review! I'm personally against mutable containers in configuration options but if you think that's the way to go, I'll do it, no problem.

I'll do it with the `getDefault...` method as you suggest. I'll also try to add some unit tests.

Reply:
Actually, now that I think about it, I'll leave the default value to be `null`. People can specify their own values if they need. Is that fine? And in that case I guess `getDefault...` method is not needed?